xcvm/architecture/transports/otp.md

```diff
@@ -0,0 +1,208 @@
+# Open Transport Protocol (OTP)
+
+Although [IBC](./ibc.md) is the default and preferred transport method within the `XCVM`, we also support arbitrary third party protocols. The abstract interface for registration and participation in `XCVM` transactions is what we call `OTP`.
+
+For a transport to be OTP compatible, it has to support the following operations:
+
+1. Fungible token transfers.
+2. Opaque data transfers.
+
+Ideally, a protocol supports both in a single operation, although most bridges will need to be wrapped in an additional set of contracts to be fully compatible.
+
+`OTP` handles the following operations:
+
+1. Maintaining a blacklist of transports.
+2. Ingress of messages on the chain.
+3. Registration of routing.
+4. Egress of messages on the chain. 
+
+## Blacklist
+
+The blacklist is used to enact emergency operations in case of bridge fraud, which can be caused by hacks, or bugs within the system. The blacklist globally disables bridges based on `LAYR` governance.
+
+Note that the blacklist is not used to manage ecosystem specific risk, such as toxic assets. It only concerns itself with bridges where the message authenticity can no longer be guaranteed.
+
+## XCVM Ingress
+
+A transport does not need to register itself to pass messages into the `XCVM`, it can do so directly, assuming that the interpreter instance accepting the transport's messages has whitelisted this transport. 
+
+The following diagram describes the flow of data (and funds) when OTP is used to for ingress of programs.
+
+```mermaid
+sequenceDiagram
+    Third-party transport->>Third-party OTP-adapter: Transform messsage.
+    Third-party OTP-adapter->>Gateway: Transfer assets and pass program.
+    Gateway->>Router: Add bridging info and transfer funds.
+    Note over Gateway: Accept/deny program based on blacklist.
```

Comment:
is Gateway singleton per chain which is governed by LAYR?

Reply:
so may be clarify what is on chain vs not in this diagram?

Reply:
Yeah, singleton (contract) per chain